index.js

```diff
@@ -116,13 +112,15 @@ const generateWord = () =>
       };
 
       console.log({ excludeSet, includeSet, excludes, includes, regexes });
-      const result = dict
-        .filter(removeDupicateLetterWords(removeDuplicates))
-        .filter(filterSet([...excludeSet, ...excludes])(false))
-        .filter(filterSet([...includeSet, ...includes])(true))
-        .filter((x) => regexes.every((pattern) => pattern.test(x)));
-
-      setSuggestion(result.random());
+      const result = addWeights(
```

Comment:
addWeights does multiple things and the name does not reflect it. Also it breaks the dot chain pipeline Preferably make it as a map function like .map(weighWords)